src/django_components/templatetags/component_tags.py

```diff
@@ -286,13 +321,32 @@ def _check_for_isolated_context_keyword(bits: List[str]) -> Tuple[List[str], boo
     return bits, False
 
 
+def _parse_tag_body(parser: Parser, end_tag: str, inline: bool) -> NodeList:
+    if inline:
+        body = NodeList()
+    else:
+        body = parser.parse(parse_until=[end_tag])
+        parser.delete_first_token()
+    return body
+
+
 def _parse_component_with_args(
```

Comment:
In this function is logic to parse args/kwargs of a component tag. I removed the part that finds the component name. Instead, this logic now lives in `TagFormatter`. So, in a way, `TagFormatter` now pre-processes the args/kwargs, extracting the component name out. And what remains, we parse as regular args/kwargs.